build/docker/intel-fpga-initcontainer.Dockerfile

```diff
@@ -1,73 +1,64 @@
 ARG DIR=/intel-device-plugins-for-kubernetes
 ARG GO111MODULE=on
+ARG BUILDFLAGS="-ldflags=-w -s"
 WORKDIR $DIR
 COPY . .
 
-RUN mkdir /install_root \
-    && swupd os-install \
-    ${CLEAR_LINUX_VERSION} \
-    --path /install_root \
-    --statedir /swupd-state \
-    --bundles=rsync \
-    --no-boot-update \
-    && rm -rf /install_root/var/lib/swupd/*
+ARG ROOT=/install_root
 
-# Build CRI Hook
-RUN cd $DIR/cmd/fpga_crihook && \
-    GO111MODULE=${GO111MODULE} go install && \
-    chmod a+x /go/bin/fpga_crihook && \
-    cd $DIR/cmd/fpga_tool && \
-    go install && \
-    chmod a+x /go/bin/fpga_tool && \
-    cd $DIR && \
-    install -D ${DIR}/LICENSE /install_root/usr/local/share/package-licenses/intel-device-plugins-for-kubernetes/LICENSE && \
-    scripts/copy-modules-licenses.sh ./cmd/fpga_crihook /install_root/usr/local/share/ && \
-    scripts/copy-modules-licenses.sh ./cmd/fpga_tool /install_root/usr/local/share/
-
-# Minimal result image
-FROM scratch as final
-COPY --from=builder /install_root /
+RUN cd cmd/fpga_crihook && GO111MODULE=${GO111MODULE} CGO_ENABLED=0 go install "${BUILDFLAGS}" && \
+    cd ../fpga_tool && GO111MODULE=${GO111MODULE} CGO_ENABLED=0 go install "${BUILDFLAGS}" && \
+    cd ../.. && \
+    install -D ${DIR}/LICENSE $ROOT/usr/local/share/package-licenses/intel-device-plugins-for-kubernetes/LICENSE && \
+    scripts/copy-modules-licenses.sh ./cmd/fpga_crihook $ROOT/usr/local/share/ && \
+    scripts/copy-modules-licenses.sh ./cmd/fpga_tool $ROOT/usr/local/share/
 
 ARG SRC_DIR=/usr/local/fpga-sw
 ARG DST_DIR=/opt/intel/fpga-sw
-
-# CRI hook
 ARG CRI_HOOK=intel-fpga-crihook
-ARG FPGA_TOOL=fpgatool
-ARG HOOK_CONF=$CRI_HOOK.json
-ARG HOOK_CONF_SRC=$SRC_DIR/$HOOK_CONF
-ARG HOOK_CONF_DST=$DST_DIR/$HOOK_CONF
 
-COPY --from=builder /go/bin/fpga_crihook $SRC_DIR/$CRI_HOOK
-COPY --from=builder /go/bin/fpga_tool $SRC_DIR/$FPGA_TOOL
+RUN install -D /go/bin/fpga_crihook $ROOT/$SRC_DIR/$CRI_HOOK
+RUN install -D /go/bin/fpga_tool $ROOT/$SRC_DIR/
 
-RUN echo -e "{\n\
+RUN echo "{\n\
     \"hook\" : \"$DST_DIR/$CRI_HOOK\",\n\
     \"stage\" : [ \"prestart\" ],\n\
     \"annotation\": [ \"fpga.intel.com/region\" ]\n\
-}\n">>$HOOK_CONF_SRC
+}\n">>$ROOT/$SRC_DIR/$CRI_HOOK.json
 
-RUN echo -e "#!/bin/sh\n\
-rsync -a --delete $SRC_DIR/ $DST_DIR\n\
-mkdir -p /etc/containers/oci/hooks.d\n\
-ln -sf $HOOK_CONF_DST /etc/containers/oci/hooks.d/$HOOK_CONF\n\
-rm $DST_DIR/deploy.sh\n\
-">> $SRC_DIR/deploy.sh && chmod +x $SRC_DIR/deploy.sh
+ARG TOYBOX_VERSION="0.8.4"
+RUN apt update && apt -y install musl musl-tools musl-dev
+RUN curl -SL https://github.com/landley/toybox/archive/refs/tags/$TOYBOX_VERSION.tar.gz | tar xz \
+    && cd toybox-$TOYBOX_VERSION \
+    && KCONFIG_CONFIG=${DIR}/build/docker/toybox-config LDFLAGS="--static" CC=musl-gcc PREFIX=$ROOT V=2 make toybox install \
+    && install -D LICENSE $ROOT/usr/local/share/package-licenses/toybox \
+    && cp -r /usr/share/doc/musl $ROOT/usr/local/share/package-licenses/
```

Comment:
This and other initcontainer dockerfiles changes are pulling source tarball from internet without checking it matches what's expected.  Shouldn't you checksum it?

Also, is there some problem in just using "busybox-static" instead: https://packages.debian.org/buster/busybox-static ?

Reply:
> Shouldn't you checksum it?

I'll add the check

Reply:
@eero-t btw, 893d859138cc0ad38c5fe18736ee372da0de8b99 is from #636

Reply:
> This and other initcontainer dockerfiles changes are pulling source tarball from internet without checking it matches what's expected. Shouldn't you checksum it?

Created issue #643 to track this